File: moxy-compiler/src/main/java/moxy/compiler/viewstateprovider/PresenterInfo.java
package moxy.compiler.viewstateprovider;

import com.squareup.javapoet.ClassName;
import javax.lang.model.element.TypeElement;

class PresenterInfo {

  private final ClassName name;

  private final ClassName viewStateName;

  PresenterInfo(TypeElement name, String viewStateName) {
    this.name = ClassName.get(name);
    this.viewStateName = ClassName.bestGuess(viewStateName);
  }

  ClassName getName() {
    return name;
  }

  ClassName getViewStateName() {
    return viewStateName;
  }
}

File: moxy-compiler/src/main/java/moxy/compiler/viewstateprovider/PresenterInfo.kt
package moxy.compiler.viewstateprovider

import com.squareup.javapoet.ClassName
import javax.lang.model.element.TypeElement

/**
 * Represents presenter class annotated with `@InjectViewState`.
 * `ViewStateProvider` will be generated based on data from this class.
 */
class PresenterInfo(val element: TypeElement, viewStateName: String) {
    val name: ClassName = ClassName.get(element)
    val viewStateName: ClassName = ClassName.bestGuess(viewStateName)
}